src/Compilers/CSharp/Portable/Emitter/Model/NamedTypeSymbolAdapter.cs

```diff
@@ -661,10 +661,13 @@ IEnumerable<Cci.IMethodDefinition> Cci.ITypeDefinition.GetMethods(EmitContext co
         {
             CheckDefinitionInvariant();
 
+            // All constructors in attributes should be emitted
+            bool isAttribute = DeclaringCompilation.IsAttributeType(this);
+
             foreach (var method in this.GetMethodsToEmit())
             {
                 Debug.Assert((object)method != null);
-                if (method.ShouldInclude(context))
+                if (isAttribute || method.ShouldInclude(context))
```

Comment:
>isAttribute [](start = 20, length = 11)

It feels unfortunate that we are going to include all attribute constructors unconditionally. Also, I am curious what other scenarios we are missing when a declaration that isn't included by default is referenced by some other API that is included. However, building a NoPia-like tracking feels like an overkill just to get the attributes case working.